src/wallet/injected/wallet-injected-interface.ts

```diff
@@ -0,0 +1,196 @@
+import {
+    FinalExecutionOutcome,
+} from '../../providers';
+import {
+    Action,
+    Transaction,
+} from '../../transaction';
+
+
+// TODOD: it should become Wallet interface, not InjectedWAllet I.
+export interface InjectedWallet {
+    /** Creates a new Function call key that is stored locally alongside with accountId*/
+    requestSignIn: ({ contractId, methodNames }: SignInOptions) => Promise<boolean>;
+
+    /** Returns true if function call key was created or false otherwise */
+    isSignedIn: () => boolean;
+
+    // TODO: Sender Wallet returns Promise<boolean>, is it possible to return boolean?
+    // TODO: Should we make signOut reurn boolean in walletconnection interface? 
+    /** Cleares function call key and accountId from local storage*/
+    signOut: () => boolean;
```

Comment:
`signOut` can be synchronous without Promise I believe. (In this mode, no popup will be shown in Sender as a reminder).